internal/encryption/config/config_load.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package config
+
+import (
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/hcl/v2/json"
+)
+
+// LoadConfigFromString loads a configuration from a string. The sourceName is used to identify the source of the
+// configuration in error messages.
+// However! this method should only be used in tests, as we should be using gohcl to parse the configuration.
+// TODO: Discuss if we can remove this now.
```

Comment:
We want to keep this here because it's great as a reference implementation. We should update the comment to make this clear.